youtube_dl/extractor/bostonglobe.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+from ..utils import (
+    extract_attributes,
+)
+
+
+class BostonGlobeIE(InfoExtractor):
+    _VALID_URL = r'https?(?i)://(?:www\.)?bostonglobe\.com/.*/(?P<id>[^/]+)/\w+(?:\.html)?'
```

Comment:
Flags like ```(?i)``` must be in the start of a regular expression since Python 3.6. See http://bugs.python.org/issue22493
```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-vF', 'test:bostonglobe']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.03.07
[debug] Git HEAD: 0f6b87d06
[debug] Python version 3.6.0 - Linux-4.10.1-1-ARCH-x86_64-with-arch
[debug] exe versions: ffmpeg 3.2.4, ffprobe 3.2.4, rtmpdump 2.4
[debug] Proxy map: {}
Traceback (most recent call last):
  File "<string>", line 23, in <module>
  File "/home/yen/Projects/youtube-dl/youtube_dl/__init__.py", line 459, in main
    _real_main(argv)
  File "/home/yen/Projects/youtube-dl/youtube_dl/__init__.py", line 449, in _real_main
    retcode = ydl.download(all_urls)
  File "/home/yen/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 1883, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/home/yen/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 752, in extract_info
    if not ie.suitable(url):
  File "/home/yen/Projects/youtube-dl/youtube_dl/extractor/common.py", line 365, in suitable
    cls._VALID_URL_RE = re.compile(cls._VALID_URL)
  File "/usr/lib/python3.6/re.py", line 233, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.6/sre_parse.py", line 856, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, False)
  File "/usr/lib/python3.6/sre_parse.py", line 415, in _parse_sub
    itemsappend(_parse(source, state, verbose))
  File "/usr/lib/python3.6/sre_parse.py", line 743, in _parse
    DeprecationWarning, stacklevel=7
DeprecationWarning: Flags not at the start of the expression https?(?i)://(?:www\ (truncated)
```